src/TabSwipe/widget/TabSwipe.ts

```diff
@@ -0,0 +1,174 @@
+import * as dojoDeclare from "dojo/_base/declare";
+import * as WidgetBase from "mxui/widget/_WidgetBase";
+import * as aspect from "dojo/aspect";
+import * as domConstruct from "dojo/dom-construct";
+import * as registry from "dijit/registry";
+
+import { SwipeHandler } from "./SwipeHandler";
+import "./ui/TabSwipe.scss";
+
+export interface TabContainer extends mxui.widget._WidgetBase {
+    declaredClass: "mxui.widget.TabContainer";
+    showTab: (tab: TabPane) => void;
+    focusIndex: number;
+    onShowTab: (callback: () => void ) => void;
+    onHideTab: (callback: () => void ) => void;
+    tabSwipeId: string;
+    validator: null;
+    _active: TabPane;
+    _tabList: HTMLElement;
+    _tabPanes: TabPane[];
+    _tabContent: TabContent;
+    _clickHandler: null;
+}
+
+export interface TabContent extends HTMLElement {
+    declaredClass: "mxui.widget.TabContent";
+}
+
+export interface TabPane extends mxui.widget._WidgetBase {
+    container: TabContainer;
+    _hidden: boolean;
+    showTab: (tab: TabPane) => void;
+    show: () => void;
+    hideTab: () => void;
+    index: number;
+    _visible: boolean;
+    visibilityIndex: number;
+}
+
+class TabSwipe extends WidgetBase {
+    private targetName: string;
+    private tabNavStyle: "tabs"| "indicators";
+    private lazyLoad: boolean;
+
+    private targetWidget?: TabContainer;
+    private targetNode: HTMLElement;
+    private swipeHandler: SwipeHandler;
+    private tabContentClass: string;
+    private swipeClass: string;
+    private targetWidgetType: string;
+
+    postCreate() {
+        this.tabContentClass = ".mx-tabcontainer-content";
+        this.swipeClass = "widget-tab-swipe";
+        this.targetWidgetType = "mxui.widget.TabContainer";
+        this.targetNode = this.findTargetNode(this.targetName);
+        this.targetWidget = this.targetNode ? registry.byNode(this.targetNode) : null;
+
+        if (!this.targetWidget) {
+            this.showError(`Tab swipe configuration error: unable to find a target with the name ${this.targetName}`);
+        } else if (this.checkCompatibility(this.targetWidget)) {
+            this.targetWidget.tabSwipeId = this.id;
+            this.targetNode.classList.add(this.swipeClass);
+        }
+    }
+
+    update(_contextObject: mendix.lib.MxObject, callback?: () => void) {
```

Comment:
Removing `_` will activate the `noUnusedParameters` rule. This is the only way to silence it, especially for overloaded functions. It's an actual typescript feature hehe...